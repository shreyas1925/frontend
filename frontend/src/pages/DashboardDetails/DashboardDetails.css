.content-entry-container {
    background-color: rgba(235, 238, 254, 255);
    height: 100vh;
    width: 80.5vw;
    position: relative;
    /* margin-top:100px */
}

.details{
    display:flex;
    flex-direction: row;
}

.content-entrie-title {
    display: flex;
    padding: 30px;
    justify-content: space-between;
}

.entrie-count {
    font-size: 30px;
    font-weight: bold;
    color: rgba(89, 89, 89, 255);
}

.new-entry-div {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 40px;
    width: 150px;
}



.new-entry-div:hover {
    cursor: pointer;
    background-color: rgba(235, 238, 254, 255);
    border-radius: 10px;
    box-shadow: 2px 3px #dddada;
}

.new-entry-text {
    font-weight: 500;
    color: rgba(57, 17, 209, 255);
}

.table-content-title {
    padding: 10px 50px 10px 50px;
    display: flex;
    margin-top: 0px;
    color: rgba(94, 94, 95, 255);
    justify-content: space-between;
}

.content-entries-fields {

    display: flex;
    justify-content: space-between;
    width: 800px;
}

.content-fields-container {
    margin-top: 20px;
    height: 60px;
    background-color: rgba(244, 246, 255, 255);
    width: 95%;
    display: flex;
    justify-content: space-between;
    align-items: center;
    border-radius: 10px;
    box-shadow: 2px 3px #dddada;
}

.icon {
    padding-left: 20px;
    height: 20px;
    width: 20px;
    margin-right: 10px;
}

.content-entries {
    display: flex;
    flex-direction: column;
    align-items: center;
}
.details--container{
    display: flex;
    height: 100vh;
    width: 100vw;
    flex-direction: row;
}

.header{
    background-color: #ffffff;
    height: 50px;
    padding: 10px;
    line-height: 50px;
}

.header h1{
    color:black;
    font-size: 25px;
    font-family: "Roboto", sans-serif;
    font-weight: 500;
    margin-left: 22px;
}

.content-entries-actions img{
    width:20px;
    height: 20px;
    margin-right: 10px;
}   

@media screen and (max-width:800px)
{
    .details{
        display: flex;
        flex-direction: column;
    }

    .content-entry-container{
        width: 100vw;
    }
}