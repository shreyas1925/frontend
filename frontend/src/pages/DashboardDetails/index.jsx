import React from 'react';
import editIcon from '../../assets/edit.png';
import binIcon from '../../assets/delete.png';
import './DashboardDetails.css';
import axios from 'axios';

import { EntryModal, Viewbar } from '../../components';

export default function DashboardDetails() {
    const [contents, setContents] = React.useState([]);

    React.useEffect(() => {
        axios
            .get('http://localhost:5000/api/getContents')
            .then((res) => {
                console.log(res);
                setContents(res.data);
            })
            .catch((err) => {
                console.log(err);
            });
    }, []);

    const [showModal, setShowModal] = React.useState(false);
    return (
        <div className='details'>
            <Viewbar contents={contents} />
            <div className='content-entry-container'>
                <div className='content-entrie-title'>
                    <span className='entrie-count'>13 Entries Found</span>
                    <div onClick={() => setShowModal(true)} className='new-entry-div'>
                        <span className='new-entry-text'>Add a new entry</span>
                    </div>
                    <EntryModal onClose={() => setShowModal(false)} show={showModal} />
                </div>
                <div className='table-content-title'>
                    <div className='content-entries-fields'>
                        <span>Id</span>
                        <span>Name</span>
                        <span>Website</span>
                        <span>Contact</span>
                    </div>
                    <span>Actions</span>
                </div>
                <div className='content-entries'>
                    <div className='content-fields-container'>
                        <div className='content-entries-fields' style={{ marginLeft: '10px' }}>
                            <span>1</span>
                            <span>John Doe</span>
                            <span>www.johndoe.com</span>
                            <span>Text</span>
                        </div>
                        <div>
                            <img className='icon' src={editIcon} alt="" />
                            <img className='icon' src={binIcon} alt="" />
                        </div>
                    </div>
                    <div className='content-fields-container'>
                        <div className='content-entries-fields' style={{ marginLeft: '10px' }}>
                            <span>1</span>
                            <span>John Doe</span>
                            <span>www.johndoe.com</span>
                            <span>Text</span>
                        </div>
                        <div>
                            <img className='icon' src={editIcon} alt="" />
                            <img className='icon' src={binIcon} alt="" />
                        </div>
                    </div>
                    <div className='content-fields-container'>
                        <div className='content-entries-fields' style={{ marginLeft: '10px' }}>
                            <span>1</span>
                            <span>John Doe</span>
                            <span>www.johndoe.com</span>
                            <span>Text</span>
                        </div>
                        <div>
                            <img className='icon' src={editIcon} alt="" />
                            <img className='icon' src={binIcon} alt="" />
                        </div>
                    </div>
                    <div className='content-fields-container'>
                        <div className='content-entries-fields' style={{ marginLeft: '10px' }}>
                            <span>1</span>
                            <span>John Doe</span>
                            <span>www.johndoe.com</span>
                            <span>Text</span>
                        </div>
                        <div>
                            <img className='icon' src={editIcon} alt="" />
                            <img className='icon' src={binIcon} alt="" />
                        </div>
                    </div>
                    <div className='content-fields-container'>
                        <div className='content-entries-fields' style={{ marginLeft: '10px' }}>
                            <span>1</span>
                            <span>John Doe</span>
                            <span>www.johndoe.com</span>
                            <span>Text</span>
                        </div>
                        <div>
                            <img className='icon' src={editIcon} alt="" />
                            <img className='icon' src={binIcon} alt="" />
                        </div>
                    </div>
                </div>
    
            </div>
        </div>
        
    );
}