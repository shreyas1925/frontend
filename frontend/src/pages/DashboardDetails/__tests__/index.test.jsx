import { BrowserRouter } from "react-router-dom";
import DashboardDetails from "..";
import { render, waitFor } from "@testing-library/react";

describe("DashboardDetails", () => {
    it("should render correctly", () => {
        const { asFragment } = render(<BrowserRouter> <DashboardDetails /></BrowserRouter>);
        expect(asFragment()).toMatchSnapshot();
    })
    it("should render and have text", async() => {
        const { getByText } = render(<BrowserRouter> <DashboardDetails /></BrowserRouter>);
        expect(getByText("loading")).toBeInTheDocument();
    })
})