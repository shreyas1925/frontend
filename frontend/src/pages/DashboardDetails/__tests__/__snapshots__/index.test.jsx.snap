// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardDetails should render correctly 1`] = `
<DocumentFragment>
   
  <div>
    <h1>
      loading
    </h1>
  </div>
</DocumentFragment>
`;
